src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx

```diff
@@ -128,7 +129,15 @@ export function SyncGasStateToSharedValues() {
 
   useEffect(() => {
     hasEnoughFundsForGas.value = undefined;
-    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || !userNativeNetworkAsset) return;
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || isLoadingNativeNetworkAsset) return;
+
+    if (!userNativeNetworkAsset) {
+      hasEnoughFundsForGas.value = false;
+      const network = getNetworkFromChainId(chainId);
+      const { nativeCurrency } = getNetworkObj(network);
+      gasAssetSymbol.value = nativeCurrency.symbol;
+      return;
+    }
```

Comment:
I'll reiterate what I put in DMs here for visibility:

Instead of reacting here in a useEffect we should only be updating the `gasAssetSymbol` when previousChainId !== currenctChainId (based off input currency).